src/game.cpp

```diff
@@ -1141,10 +1141,16 @@ bool game::do_turn()
 
     // Recover some stamina every turn.
     if( u.stamina < u.get_stamina_max() && !u.has_effect("winded") ) {
-        // But mouth encumberance interferes.
+        // But mouth encumbrance interferes.
         u.stamina += std::max( 1, 10 - (u.encumb(bp_mouth) / 10) );
         // TODO: recovering stamina causes hunger/thirst/fatigue.
     }
+    
+    // 3d4 bonus stamina from active stimpack stamina-boost.
+    if( u.stamina < u.get_stamina_max() && u.has_effect("stimpack") && 
+        u.get_effect_dur("stimpack") > 50 ) {
+        u.stamina += std::max( 3, 12 );
```

Comment:
As we're starting to add synthetic performance-enhancing drugs, I'd love to
start removing the nonsense buffs from recreational drugs.